include/GoogleSync/GoogleSync.php

```diff
@@ -209,13 +241,24 @@ public function getClient($id)
 
         // Refresh the token if needed
         if ($client->isAccessTokenExpired()) {
-            $GLOBALS['log']->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Refreshing Access Token');
+            $this->logger->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Refreshing Access Token');
             $refreshToken = $client->getRefreshToken();
-            $client->fetchAccessTokenWithRefreshToken($refreshToken);
+            if (empty($refreshToken)) {
+                throw new Exception('Refresh token is missing');
+                return false;
+            } else {
+                try {
```

Comment:
Too many nested block. - After a thow or return, the `else` became unnecessary.

Reply:
Fixed